From 86e458494effa7e9b174c32a5bbe9ea0a141204b Mon Sep 17 00:00:00 2001
From: Alexandre Bailon <abailon@baylibre.com>
Date: Thu, 23 Feb 2023 10:52:57 +0100
Subject: [PATCH] greybus: netlink: Fix a netlink warning and double free issue

In case of error, genlmsg_unicast already cancel the skb and free it.
Return the error code and don't cancel and free the skb.

Signed-off-by: Alexandre Bailon <abailon@baylibre.com>
---
 drivers/greybus/netlink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/greybus/netlink.c b/drivers/greybus/netlink.c
index 56f507ebdd0b..75010952a789 100644
--- a/drivers/greybus/netlink.c
+++ b/drivers/greybus/netlink.c
@@ -140,7 +140,7 @@ static int message_send(struct gb_host_device *hd, u16 cport_id,
 
 	retval = genlmsg_unicast(&init_net, skb, GB_NL_PID);
 	if (retval)
-		goto err_cancel;
+		return retval;
 
 	greybus_message_sent(hd, message, 0);
 
-- 
2.39.1

